akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -37,6 +39,143 @@ private[stream] object AbstractStage {
   final val NoTerminationPending = 0x8000
   final val BothBalls = UpstreamBall | DownstreamBall
   final val BothBallsAndNoTerminationPending = UpstreamBall | DownstreamBall | NoTerminationPending
+
+  private class PushPullGraphLogic[In, Out, Ext](
+    private val shape: FlowShape[In, Out],
+    val attributes: Attributes,
+    val stage: AbstractStage[In, Out, Directive, Directive, Context[Out], LifecycleContext])
+    extends GraphStageLogic(shape) with AsyncContext[Out, Ext] {
+
+    final override def materializer: Materializer = interpreter.materializer
+
+    // FIXME: Can be removed from here and from Context
+    override private[stream] def enter(): Unit = ???
```

Comment:
these are no longer needed, I can remove them as soon as I remove the old interpreter.